table/block_based/block_based_table_reader.cc

```diff
@@ -500,10 +501,11 @@ Status GetGlobalSequenceNumber(const TableProperties& table_properties,
   }
 
   uint32_t version = DecodeFixed32(version_pos->second.c_str());
-  if (version < 2) {
+  if (version != 2) {
     if (seqno_pos != props.end() || version != 1) {
```

Comment:
Won't version 3+ fail no matter what?

Reply:
I guess they should, but the error message about global seqno is confusing.

Reply:
Yes. We only have version 1 and 2 https://github.com/facebook/rocksdb/blob/0fca5e31b4564ef36960bfa487ebbd33c49f8445/table/sst_file_writer.cc#L367
so I want to fail invalid versions here.

Reply:
Let me update the error message.